public/app/plugins/datasource/loki/configuration/DerivedFields.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState } from 'react';
+import { css } from 'emotion';
+import { Button, DataLinkBuiltInVars, stylesFactory, useTheme, VariableOrigin } from '@grafana/ui';
+import { GrafanaTheme } from '@grafana/data';
+import { DerivedFieldConfig } from '../types';
+import { DerivedField } from './DerivedField';
+import { DebugSection } from './DebugSection';
+
+const getStyles = stylesFactory((theme: GrafanaTheme) => ({
+  infoText: css`
+    padding-bottom: ${theme.spacing.md};
+    color: ${theme.colors.textWeak};
+  `,
+  derivedField: css`
+    margin-bottom: ${theme.spacing.sm};
+  `,
+}));
+
+type Props = {
+  value?: DerivedFieldConfig[];
+  onChange: (value: DerivedFieldConfig[]) => void;
+};
+export const DerivedFields = (props: Props) => {
+  const { value, onChange } = props;
+  const theme = useTheme();
+  const styles = getStyles(theme);
+
+  const [showDebug, setShowDebug] = useState(false);
+
+  return (
+    <>
+      <h3 className="page-heading">Derived fields</h3>
+
+      <div className={styles.infoText}>
+        Derived fields can be used to extract new fields from the log message and create Data Link from it's value.
+      </div>
+
+      <div className="gf-form-group">
+        {value &&
+          value.map((field, index) => {
+            return (
+              <DerivedField
+                className={styles.derivedField}
+                key={index}
+                value={field}
+                onChange={newField => {
+                  const newDerivedFields = [...value];
+                  newDerivedFields.splice(index, 1, newField);
+                  onChange(newDerivedFields);
+                }}
+                onDelete={() => {
+                  const newDerivedFields = [...value];
+                  newDerivedFields.splice(index, 1);
+                  onChange(newDerivedFields);
+                }}
+                suggestions={[
```

Comment:
The suggestion seems a bit clumsy, but I guess we're trying to be consistent here with the data links API. But we're then limited to extracting only one match group, right?

Reply:
Yes one thing is consistency and second is that we can add more things you could use here. Like references to other fields or timestamp or reference to query. Right now the DataLinks themselves use more variables but whether some of the are actually interpolated depends on where you click in a graph so there are some inconsistencies and I would rather add more variables later rather than add stuff that may not work. Also referencing other fields value, which I think may make the most sense here is not implemented yet in the DataLinks.